src/operator/nn/cudnn/cudnn_algoreg-inl.h

```diff
@@ -44,6 +44,10 @@ namespace op {
  */
 template <typename CuDNNAlgoType>
 class CuDNNAlgo {
+  // Compile guard needed to get past non-nvcc compilation of cudnn_algoreg.cc.
```

Comment:
Why?

Reply:
I've pushed a fix that removes the compile guard.  Before the fix and without the guard, the following step of the compile failed:
```
g++ ... -DMSHADOW_USE_CUDNN=1 ... -c cudnn_algoreg.cc
```
The problem is that the cudnn_algoreg-inl.h and cudnn_algoreg.cc files look at MXNET_USE_CUDNN, not MSHADOW_USE_CUDNN.  The header file ./include/mxnet/libinfo.h has the vital missing line that equates the two:
```
#define MXNET_USE_CUDNN MSHADOW_USE_CUDNN
```
So this was a case of not adhering to the rule 'include what you use.'  The header file ./src/common/cuda_utils.h was referring to MXNET_USE_CUDNN, but not including any header file that would ensure libinfo.h was included.  Now fixed.

Now that MSHADOW is part of MXNET and no longer a submodule, is there a benefit to consolidating the dual naming as in {MSHADOW,MXNET}_USE_CUDNN?